src/main/kotlin/graphics/scenery/geometry/Curve.kt

```diff
@@ -242,91 +244,150 @@ class Curve(spline: Spline, partitionAlongControlpoints: Boolean = true, private
 
     companion object VerticesCalculation {
         /**
-         * This function calculates the triangles for the the rendering. It takes as a parameter
+         * This function calculates the triangles for the rendering. It takes as a parameter
          * the [curveGeometry] List which contains all the baseShapes transformed and translated
          * along the curve.
          */
-        fun calculateTriangles(curveGeometry: List<List<Vector3f>>, addCoverOrTop: Int = 2): ArrayList<Vector3f> {
+        fun calculateTriangles(curveGeometry: List<List<Vector3f>>, addCoverOrTop: Int = 2): Pair<ArrayList<Vector3f>, ArrayList<Vector3f>> {
             val verticesVectors = ArrayList<Vector3f>(curveGeometry.flatten().size * 6 + curveGeometry[0].size + 1)
+            val normalVectors = ArrayList<Vector3f>(verticesVectors.size/3)
```

Comment:
This is unfortunately wrong - there has to be one normal vector for each vertex.

Reply:
@skalarproduktraum  Check, I did not know how to compute the normals properly.
A valid approach seems to be to use the normals of the triangles surrounding a vertex, add them together, and normalize the final results. First question: is that correct?
If yes, then there is definitely some bug still somewhere because we went from zebra to a full-blown rip look, see below
![CurveUniformBSplineExample - 2022-11-15 22 58 27](https://user-images.githubusercontent.com/45595982/202041952-ce400ff9-7e15-42c9-b1c6-78b5e99ca12f.png)